aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -563,6 +563,19 @@ Tensor scatter_add(const Tensor & self, int64_t dim, const Tensor & index, const
   return self.clone(at::MemoryFormat::Preserve).scatter_add_(dim, index, source);
 }
 
+Tensor scatter_add_src_backward(
+  const Tensor& grad, int64_t dim, const Tensor& index,
+  IntArrayRef src_sizes, bool sparse_grad
+) {
+  auto grad_src = at::zeros(src_sizes, grad.options()).contiguous();
+  auto grad_contiguous = grad.contiguous();
```

Comment:
I wonder if it is all right to make gradients contiguous...

Reply:
It is all right yes.
Needed, I don't know. Does gather_stub works fine with non-contiguous source?

Reply:
If TensorIterator works with non-contigous tensors, then gather_stub should work as well, I guess. Could you please explain what exactly non-contiguity means?

Reply:
In this case it would be Tensors with funky strides.
And yes I think TensorIterator should handle that just fine.